src/server/modes/charts/plugins/datalens/preparers/line/helpers/line-mappers/mapDataToDimensionColoredLines.ts

```diff
@@ -164,13 +167,22 @@ export const mapDataToDimensionColoredLines = ({
             combinedValue: itemValues.extraValue,
         });
 
-        line.colorValue = colorValue;
+        line.colorValue = colorModeIsGradient ? Number(colorValue) : colorValue;
         line.shapeValue = shapeValue;
     }
 
     const lastKey = typeof xValue === 'undefined' ? shownTitle : xValue;
 
-    lines[key].data[lastKey as string | number] = {value: yValue};
+    const pointData: {
+        value: typeof yValue;
+        colorValue?: number;
+    } = {value: yValue};
+
+    if (colorModeIsGradient) {
+        pointData.colorValue = Number(lines[key].colorValue);
```

Comment:
There is already a cast to Number above. Do you need to duplicate here?

Reply:
We cast `line.colorValue` above. And here we cast `point.colorValue`.